server/routerlicious/packages/services-telemetry/src/lumberFormatters.ts

```diff
@@ -0,0 +1,40 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Lumberjack } from "./lumberjack";
+import { ILumberFormatter, LogLevel } from "./resources";
+import { Lumber } from "./lumber";
+
+export class SanitizationLumberFormatter implements ILumberFormatter {
+	private readonly sensitiveKeys = [
+		/cookie/i,
+		/passw(or)?d/i,
+		/^pw$/,
+		/^pass$/i,
+		/secret/i,
+		/token/i,
+		/api[._-]?key/i,
+		/session[._-]?id/i,
+	];
+
+	private readonly redactedStr = "[LUMBER_REDACTED]";
+
+	public transform(lumber: Lumber<string>): void {
+		if (lumber.logLevel === LogLevel.Error && lumber.exception) {
```

Comment:
I think the formatter should work traverse the exception object rather than the Lumber object. Also, this check for error/exception can be moved up to Lumber.Emit(). Of course, that will limit the application of formatters to error messages only, but I think we won't like to apply formatters on anything other than errors, given the potential perf implications. Thoughts?

Reply:
This was my bad, I always intended on traversing through the exception. Modified.